research_town/agents/agent.py

```diff
@@ -28,12 +44,14 @@ def __init__(
         self,
         profile: Profile,
         model_name: str,
+        log_db: LogDB,
```

Comment:
agent should not be linked with DB. this is dangerous design thinking from users' perspective